day29/C++/kesh5210_binary_search.cpp

```diff
@@ -0,0 +1,83 @@
+/**
+* @author: Hrishikesh-Thakkar
+* @date: 10/1/2020
+**/
+
+#include<bits/stdc++.h>
+using namespace std;
+
+//Recursive Approach
+int binarySearchRecurse(vector<int> &arr,int low,int high,int key){
+
+	if(low<=high){
+		int mid = low + (high-low)/2;
+		if(arr[mid] == key) {
+			return mid;
+		}
+		else if(arr[mid]<key) {
+			return binarySearchRecurse(arr,mid+1,high,key);
+		}
+		else {
+			return binarySearchRecurse(arr,low,mid-1,key);
+		}
+	}
+	return -1;
+}
+
+//Iterative Approach
+int binarySearchIterate(vector<int> &arr,int low, int high, int key){
+
+	while(low<=high){
+		int mid = low + (high-low)/2;
+		if(arr[mid] == key){
+			return mid;
+		}
+		else if(arr[mid]<key){
+			low = mid+1;
+		}
+		else{
+			high = mid-1;
+		}
+	}
+
+	return -1;
+}
+int main(){
+
+	//Input array
+	vector<int> arr;
+	for(int i=0;i<=8;i++)
+		arr.push_back(pow(2,i));
+
+	//Number to search for
+	int key = 32;
+
+	//In case Array is not sorted as Binary Search works only with Sorted Arrays

```

Comment:
Since you have created an array of powers of 2 starting from left to right, isn't it sorted in an ascending order?

Reply:
Yes that's correct, that's why I added the comment that incase it's not sorted. To illustrate that it doesn't work with any array configuration, only sorted ones.

Reply:
This comment is confusing because it makes it seem as though the current array is not sorted. I'd recommend either leaving it out or adding an example where the array is unsorted before adding the comment.

Reply:
Alright, I'll make that change.